src/middleware.rs

```diff
@@ -89,10 +91,10 @@ impl<S, B, RootSpan> Transform<S, ServiceRequest> for TracingLogger<RootSpan>
 where
     S: Service<ServiceRequest, Response = ServiceResponse<B>, Error = Error>,
     S::Future: 'static,
-    B: 'static,
-    RootSpan: RootSpanBuilder,
+    B: MessageBody + 'static,
+    RootSpan: RootSpanBuilder + Unpin,
 {
-    type Response = ServiceResponse<B>;
+    type Response = ServiceResponse<StreamSpan<B>>;
```

Comment:
Since `StreamSpan<B>` implements `MessageBody` where `B: StreamSpan`, this reads as a plain `ServiceResponse<B: MessageBody>` to any possible callers

Reply:
Does that mean that (just like `Logger`) we might need the [compat middleware](https://docs.rs/actix-web/4.0.0-beta.9/actix_web/middleware/struct.Compat.html) in some cases ? Do we want to say something about it on the documentation ?

Reply:
Yes, it looks like the compat middleware would be necessary for mounting inside scopes, resources, and conditions with this change

Reply:
So you're right that there should be documentation about this. I'll see if I can write some up